foreman.te

```diff
@@ -171,190 +188,156 @@ require{
     type cockpit_session_exec_t;
 }
 
-#######################################
+######################
 #
-# Passanger/httpd local policy
+# Ruby on Rails policy
 #
 
-# https://bugzilla.redhat.com/show_bug.cgi?id=1716944
-allow passenger_t self:capability dac_override;
-allow passenger_t self:capability sys_resource;
-allow passenger_t self:process signull;
-allow passenger_t self:tcp_socket listen;
+# Service transition
+init_daemon_domain(foreman_rails_t, foreman_rails_exec_t)
 
-# rubygem FFI is used in foreman-tasks (BZ#1670109 / RM#26951)
-allow passenger_t self:process execmem;
+# Generic domain rules
+auth_read_passwd(foreman_rails_t)
+dev_list_sysfs(foreman_rails_t)
+dev_read_sysfs(foreman_rails_t)
+hostname_exec(foreman_rails_t)
+ipa_filetrans_named_content(foreman_rails_t)
+kernel_read_network_state(foreman_rails_t)
+kernel_read_system_state(foreman_rails_t)
+sysnet_read_config(foreman_rails_t)
 
-miscfiles_read_localization(passenger_t)
+# Certificates (Foreman uses puppet certificates as well)
+miscfiles_read_certs(foreman_rails_t)
+puppet_read_config(foreman_rails_t)
 
-# Allow Foreman to connect to Foreman Proxy on port 9090 (Katello)
-# or to connect to remote Cockpit instance (via Remote Execution)
-allow passenger_t websm_port_t:tcp_socket name_connect;
+# It is not possible to do transition into SCL application
+# without calling a shell: https://github.com/sclorg/scl-utils/issues/31
+# These rules are for: /usr/libexec/foreman/tfm-ruby-selinux
+corecmd_exec_shell(foreman_rails_t)
+corecmd_exec_ls(foreman_rails_t)
+# The following is needed for "scl enable" which creates temporary
+# shell script /var/tmp/xxxxxxx and a subprocess.
+files_manage_generic_tmp_files(foreman_rails_t)
```

Comment:
Is this still the case with our `scl_source` + exec wrappers? We dropped `/usr/libexec/foreman/tfm-ruby-selinux`

Reply:
Unfortunately yes, I was hoping that this will be not necessary but I still see a denial. Honestly, don't know which component does that, some subprocess? I stopped digging when I learned that we cannot live with the other two rules above which are far more dangerous than tmp files read write. So I suggest to live with this.

Reply:
I'm guessing `scl_source` generates that then.

Should we hide that behind an EL7 conditional since EL8 doesn't use SCLs?